qiskit/quantum_info/states/statevector.py

```diff
@@ -862,20 +862,31 @@ def _evolve_instruction(statevec, obj, qargs=None):
             return statevec
         if isinstance(obj, Barrier):
             return statevec
-        if isinstance(obj, Initialize):
+        if isinstance(obj, Initialize) and qargs is None:
             # state is initialized to labels in the initialize object
             if all(isinstance(param, str) for param in obj.params):
-                statevec._data = Statevector.from_label("".join(obj.params))._data
+                initialization = Statevector.from_label("".join(obj.params))._data
             # state is initialized to an integer
             # here we're only checking the length as (1) a length-1 object necessarily means the
             # state is described by an integer (as labels were already covered) and (2) the int
             # was cast to a complex and we cannot do an int typecheck anyways
             elif len(obj.params) == 1:
                 state = int(np.real(obj.params[0]))
-                statevec._data = Statevector.from_int(state, (2,) * obj.num_qubits)._data
+                initialization = Statevector.from_int(state, (2,) * obj.num_qubits)._data
             # state is initialized to the statevector
             else:
-                statevec._data = np.asarray(obj.params, dtype=complex)
+                initialization = np.asarray(obj.params, dtype=complex)
+
+            if qargs is None:
```

Comment:
In a similar note to @ikkoham, if line 865 is true, then line 880 is always true, right?

Reply:
Whoops, that check on line 865 was left from a test, that shouldn't be there!